From 36804f864cbb079bc2164ff40b8cda500da20296 Mon Sep 17 00:00:00 2001
From: ghsr <ghsr92@yandex.ru>
Date: Tue, 27 Sep 2016 19:44:42 +0600
Subject: [PATCH] Revert "wifi: update the insmod function to use finit_module"

This reverts commit 02bd5eb704c528eaa61a0043404566fb88d17a90.

 Conflicts:
	wifi/wifi.c

Change-Id: If72f4f01d1acc1dd973a37cdb908687e7f3eac4e
---
 wifi/wifi.c | 25 ++++++++++++-------------
 1 file changed, 12 insertions(+), 13 deletions(-)

diff --git a/wifi/wifi.c b/wifi/wifi.c
index d92cdbd..8b5f061 100644
--- a/wifi/wifi.c
+++ b/wifi/wifi.c
@@ -23,7 +23,6 @@
 #include <sys/stat.h>
 #include <unistd.h>
 #include <poll.h>
-#include <sys/syscall.h>
 
 #ifdef USES_TI_MAC80211
 #include <dirent.h>
@@ -202,19 +201,19 @@ char* get_samsung_wifi_type()
 
 int insmod(const char *filename, const char *args)
 {
-     /* O_NOFOLLOW is removed as wlan.ko is symlink pointing to
-        the vendor specfic file which is in readonly location */
-     int fd = open(filename, O_RDONLY | O_CLOEXEC);
-     if (fd == -1) {
-        ALOGD("insmod: open(\"%s\") failed: %s", filename, strerror(errno));
+    void *module;
+    unsigned int size;
+    int ret;
+
+    module = load_file(filename, &size);
+    if (!module)
         return -1;
-     }
-     int rc = syscall(__NR_finit_module, fd, args, 0);
-     if (rc == -1) {
-       ALOGD("finit_module for \"%s\" failed: %s", filename, strerror(errno));
-     }
-     close(fd);
-     return rc;
+
+    ret = init_module(module, size, args);
+
+    free(module);
+
+    return ret;
 }
 
 int rmmod(const char *modname)
-- 
2.9.3

